frontend/static_vol/js/modules/tournamentList.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+import { labels } from "./labels.js";
+import { fetchTournaments } from "./tounamentApi.js";
+import { formatDateToLocal } from "./formatDateToLocal.js";
+import { resetListenUpcomingTournamentList } from "./tournamentListListener.js";
+
+const updateUpcomingTournamentList = async (pageInstance) => {
+    try {
+        const tournaments = await fetchTournaments('upcoming');
+        const listWrapper = document.querySelector('.blockTournamentList_upcoming');        
+        if (tournaments.length === 0) {
+            listWrapper.innerHTML = `<p>${'No upcoming tournament'}</p>`;
+//            listWrapper.innerHTML = `<p>${labels.tournament.msgNoUpcoming}</p>`;
+        } else {
+            listWrapper.innerHTML = '';
+            tournaments.forEach(tournament => {
+                const formatedStartDate = formatDateToLocal(tournament.start);
+                let nicknameHtml = '';
+                let buttonHtml = '';
+                if (tournament.nickname != '') {
+                    nicknameHtml = `<p class="unitTournament_nickname">as ${tournament.nickname}</p>`;
+                    buttonHtml = `<p class="blockForm_button"><button type="submit" class="unitUpcomingTournamentButton_cancel unitButtonDecline" data-name="${tournament.name}" data-id="${tournament.id}">${labels.tournament.labelCancelEntry}</button></p>`;                   
+                } else if (tournament.current_participants >= tournament.max_participants) {
+                    buttonHtml = `<p class="blockForm_button"><button type="button" disabled>FULL</p>`;
+                //    buttonHtml = `<p class="blockForm_button"><button type="button" disabled>${labels.tournament.labelFull}</p>`;
+                } else if (tournament.nickname === '') {
+                    buttonHtml = `<p class="blockForm_button"><button type="button" class="unitUpcomingTournamentButton_entry unitButton" data-name="${tournament.name}" data-id="${tournament.name}}">${labels.tournament.labelEntry}</button></p>`;
```

Comment:
```suggestion
                    buttonHtml = `<p class="blockForm_button"><button type="button" class="unitUpcomingTournamentButton_entry unitButton">${labels.tournament.labelEntry}</button></p>`;
```
ここは`<input type="hidden" 〜>` からデータ取得しているのでdata属性なしで大丈夫そうです

Reply:
ありがとうございます！